djangocms_versioning/cms_toolbars.py

```diff
@@ -131,6 +144,87 @@ def post_template_populate(self):
         self._add_versioning_menu()
 
 
+class VersioningPageToolbar(PageToolbar):
+    """
+    Overriding the original why???
+
+    The default language menu contains the
+    """
+    def get_page_content(self, language=None):
+        if not language:
+            language = self.current_lang
+        page_content = PageContent._original_manager.filter(
+            page=self.page, language=language
+        ).first()
+        return page_content or None
+
+    def populate(self):
+        self.page = self.request.current_page or getattr(self.toolbar.obj, "page", None)
+        self.title = self.get_page_content() if self.page else None
+        self.permissions_activated = get_cms_setting('PERMISSION')
+
+        self.override_language_menu()
+        self.change_admin_menu()
+        self.add_page_menu()
+        self.change_language_menu()
+
+    def override_language_menu(self):
+        """
+        Override the default language menu for pages that are versioned
+        """
+        # Only override the menu if a page can be found
+        if settings.USE_I18N and self.page:
+            language_menu = self.toolbar.get_menu(LANGUAGE_MENU_IDENTIFIER, _('Language'))
+
+            # remove_item uses `items` attribute so we have to copy object
+            for _item in copy(language_menu.items):
+                language_menu.remove_item(item=_item)
+
+            for code, name in get_language_tuple(self.current_site.pk):
+                # Get the content, it could be draft too hence using _original_manager!
+                page_content = self.get_page_content(language=code)
+                if page_content:
+                    url = get_object_preview_url(page_content, code)
+                    language_menu.add_link_item(name, url=url, active=self.current_lang == code)
```

Comment:
This method replaces the old urls with new urls that are guaranteed to be able to fetch drafts. I don't think this would invalidate the following PR: https://github.com/divio/django-cms/pull/6826

I would  say that the testcase I added in that PR should pass because that is a behaviour we want even without versioning. So just keeping that in mind. 

The same goes for this PR: https://github.com/divio/django-cms/pull/6828 the links should be correct without versioning as well.

Reply:
I think this PR would mean the following is no longer required because there is no such thing as draft in standalone: https://github.com/divio/django-cms/pull/6826/files


I think that this PR is still valid because the url generated is always using the currnet page and not the language inside the loop: https://github.com/divio/django-cms/pull/6828/files

Reply:
Regarding 6826, I'd argue for the sake of consistency to keep it the same; ie preview-url in core, preview-url in versioning.

Reply:
Ok, I agree on that for consistency. I'll comment in that PR so it's not lost.